frame/composable-traits/src/vamm.rs

```diff
@@ -11,18 +15,27 @@ use frame_support::pallet_prelude::DispatchError;
 pub trait Vamm {
 	/// The balance type for an account.
 	type Balance;
-
+	/// Signed fixed point number implementation
+	type Decimal: FixedPointNumber;
+	/// Configuration for creating and initializing a new vAMM instance. May be used in extrinsic
+	/// signatures
+	type VammConfig: FullCodec + MaxEncodedLen + TypeInfo + Debug + Clone + PartialEq;
 	/// The identifier type for each virtual automated market maker.
-	type VammId;
+	type VammId: FullCodec + MaxEncodedLen + TypeInfo;
```

Comment:
@Cardosaum lemme know if you disagree with this solution. What motivates this is:
* the associated `VammConfig` type provides a little more generality for its implementation
* the Clearing House doesn't care about the exact attributes of `VammConfig`; it is created and provided by the extrinsic caller via the `MarketConfig` struct
* the Clearing House **does** care about the trait bounds of `VammConfig` here, since it will be indirectly part of an extrinsic signature

This also allows other implementers of the `Vamm` trait to have their own `VammConfig`s. If for any reason some code requires the trait to use the `VammConfig` defined in this file, it can do so by specifying the bound
```rust
type Vamm: Vamm<VammConfig = VammConfig<BalanceType>>;
```

Reply:
Yeah, sounds good. If for any reason that changes in the future, we can always remove the bound